File: AngularClient/AngularClient/src/Services/UserService.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { User } from "src/models/user";

@Injectable()
export class UserServices {
    constructor(private _http: HttpClient) { }
    getAllUsers() {
        return this._http.get(`https://localhost:7164/api/users/GetAllUsers`).toPromise();
    }

    getUser(login: string, password: string) {
        return this._http.get(`https://localhost:7164/api/users/GetUser?login=${login}&password=${password}`).toPromise();
    }

    updateUser(user: User) {
        return this._http.put(`https://localhost:7164/api/users/UpdateUser`, user).toPromise()
    }

    addUser(user: User) {
        return this._http.post(`https://localhost:7164/api/users/AddUser`, user).toPromise();
    }

    deleteUser(userId: number) {
        return this._http.get(`https://localhost:7164/api/users/DeleteUser?id=${userId}`).toPromise();
    }
}

File: WebClientMVC/Services/UserService.cs
using Api.Data;
using Api.Data.Models;
using WebClientMVC.Repository;

namespace WebClientMVC.Services
{
    public class UserService
    {
        private readonly UserRepository _repository = new UserRepository();

        public Task<User?> GetUser(string login, string password)
        {
            return _repository.GetUser(login, password);
        }

        public Task<List<User>?> GetAllUsers()
        {
            return _repository.GetAllUsers();
        }

        public Task<User?> AddUser(string login, string password)
        {
            return _repository.AddUser(login, password);
        }

        public Task<User?> UpdateUser(User user)
        {
            return _repository.UpdateUser(user);
        }

        public Task<bool?> DeleteUser(User user)
        {
            return _repository.DeleteUser(user);
        }
    }
}
